### ISSUE TITLE
Dimensional Collapse Tracking

### ISSUE THREAD

**Issue by zankerx** [facebookresearch/jepa#77]
Hello, first of all, thank you for sharing this kind of model.

I have a question regarding the development of this model, particularly about the collapse of the dimension. Do you use any other indicators besides loss during training to select hyper-parameters and maximize the amount of information contained in the embedding? (Knn, eigenvalues of the embedding, or others?).

Thank you in advance for your response :)

**Comment by vimalthilak** [comment]
@zankerx there is a line of research on embedding quality that maybe useful for what you are thinking about. Metrics such as \alpha-Req, RankMe or LiDAR (self-promotion alert as I am one of the authors), CLID have shown to be useful for estimating embedding quality in image-based  SSL (and transfer learning) methods

**Comment by c3-utsavdutta98** [comment]
@vimalthilak great work with Lidar! was curious if you guys were able to empirically validate how Lidar correlates with non-linear probing? For reference, VJEPA performs significantly better with a **non-linear attentive pooler** on downstream tasks, and was curious if you were able to evaluate how well LIDAR correlates to probes of a similar nature (even something simpler like rbf kernel SVMs for example)

**Comment by vimalthilak** [comment]
@c3-utsavdutta98 great question! You are correct on both the fact that our work empirically tested linear probing (although the linear heads typically have a non-learnable batchnorm layer followed by linear layer) while V-JEPA uses attentive pooling + linear layer. I'll add the above as a work item although I can't provide a timeline for results due to other priorities. If you have any experience I'm very eager to learn about your findings.

**Comment by c3-utsavdutta98** [comment]
Hi Vimal! Thanks for the response.

We haven't done as thorough an empirical study as yours, but have used LiDAR as a proxy metric to track while training JEPA based architectures, with a mixture of downstream classifiers (both linear and non-linear). The correlation with a downstream support vector classifier is not as clean as the results for liner probing, although I suspect that is expected, as a higher rank embedding space would make finding linear subspaces for classification easier, but it is tough to say the same thing about non-linear methods.

Would love to setup a more rigorous set of experiments to validate these though, as it would be an extremely useful summary metric to track SSL based methods, especially self-distillation based ones, where the training loss is itself quite uninformative.
